src/sentry/users/api/serializers/authenticator.py

```diff
@@ -0,0 +1,100 @@
+from collections.abc import Mapping
+from datetime import datetime
+from typing import Any, TypedDict, cast
+
+from sentry.api.serializers import Serializer, register
+from sentry.auth.authenticators.base import AuthenticatorInterface
+from sentry.auth.authenticators.recovery_code import RecoveryCodeInterface
+from sentry.auth.authenticators.sms import SmsInterface
+from sentry.auth.authenticators.totp import TotpInterface
+from sentry.auth.authenticators.u2f import U2fInterface
+from sentry.users.models.user import User
+
+
+class EnrolledAuthenticatorInterfaceSerializerResponse(TypedDict, total=False):
+    authId: str
+    createdAt: datetime
+    lastUsedAt: datetime | None
```

Comment:
Another way to solve this in future changes would be to use `typing.NotRequired` to mark these keys as optional.

Reply:
actually that's a good point -- is this actually optional and nullable? (rarely something is)

Reply:
@asottile-sentry like are the "enrolled" fields optional & nullable ? I believe so, because they are some test cases like `tests/sentry/users/api/endpoints/test_user_authenticator_enroll.py` where the enrollment in an authenticator interface failed, so the returned user doesn't have any of these fields. Or was something else nullable ? ...

Reply:
as written now there's these possibilities:
```
{}
{"lastUsedAt": some_datetime}
{"lastUsedAt": None}
```

which is usually a bug

Reply:
yeah I dunno, from what I'm seeing from our tests
`{}` -> is a real scenario [it seems like this can happen when enrollment fails], so Optional seems to be valid
`{"lastUsedAt": some_datetime}` -> also is a real scenario when we mark an authenticator as used.
`{"lastUsedAt": None}` -> also is real when a user is newly enrolled

It looks like the three scenarios all happen so probably not a bug, albeit also somewhat confusing to deal with